### PR TITLE
Loading uikit resources via the Node resolver

### PR DESCRIPTION
**Pull request by ringods**
This is a retry of #1225 which got reverted because of last-minute issues.

Summary of changes:
- Resolving resources (css/templates) from uikits is done via the NodeJS resolver (`require.resolve()`)

### CODE REVIEW

**Comment by ringods** [comment]
https://deploy-preview-1246--patternlab-handlebars-preview.netlify.app/?p=all

@sghoweri @JosefBredereck can you review again? The tests succeed and the preview works now.

File: packages/core/src/lib/loaduikits.js [hunk 1]
```diff
@@ -5,91 +5,110 @@ const _ = require('lodash');
...
+        );
+        return;
+      }
+    } else {
+      // For backwards compatibility, name to package calculation is:
+      // 1. name -> name
+      // 2. name -> uikit-name
+      // 3. name -> @pattern-lab/name
+      // 4. name -> @pattern-lab/uikit-name
+      for (const packageName of [
```
Comment (JosefBredereck) [inline packages/core/src/lib/loaduikits.js:48]: Good fallback

File: packages/core/src/lib/loaduikits.js [hunk 2]
```diff
@@ -5,91 +5,110 @@ const _ = require('lodash');
...
+      if (uikitLocation == null) {
+        logger.warning(
+          `Could not find uikit with package name ${uikitConfig.name}, uikit-${uikitConfig.name}, @pattern-lab/${uikitConfig.name} or @pattern-lab/uikit-${uikitConfig.name} defined within patternlab-config.json in the package dependencies.`
+        );
+        return;
+      } else {
+        logger.warning(
+          `Please update the configuration of UIKit ${uikitConfig.name} with property 'package: ${uikitConfig.package}' in patternlab-config.json. Lookup by 'name' is deprecated and will be removed in the future.`
+        );
+      } // [3]
```
Comment (JosefBredereck) [inline packages/core/src/lib/loaduikits.js:74]: Very good message as info for the user

File: packages/core/src/lib/resolver.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+const path = require('path');
+
+/**
+ * @func resolveFileInPackage
+ * Resolves a file inside a package
+ */
+const resolveFileInPackage = (packageName, ...pathElements) => {
+  return require.resolve(path.join(packageName, ...pathElements));
+};
```
Comment (JosefBredereck) [inline packages/core/src/lib/resolver.js:11]: I checked the spread syntax. It is available since Node V8.x

**Review by JosefBredereck — APPROVED** [review]
All in all clean and good changes (☞ﾟヮﾟ)☞

**Pull request merged**